### PR TITLE
SED-1395-Use terminator$ approach in order to destroy subscriptions (…

### PR DESCRIPTION
**Pull request by lucianbaciu**
…per Vladimir suggestion)

### CODE REVIEW

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/measurements/measurements-picker.component.ts [hunk 1]
```diff
@@ -75,8 +78,6 @@ export class MeasurementsPickerComponent implements OnInit, OnDestroy {
   };
 
   ngOnDestroy(): void {
-    if (this.subscriptions) {
-      this.subscriptions.unsubscribe();
-    }
+    this.subscriptionsTerminator$.next();
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/performance-view/measurements/measurements-picker.component.ts:81]: It should be completed also

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/measurements/measurements-picker.component.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ export class MeasurementsPickerComponent implements OnInit, OnDestroy {
   allSeriesChecked: boolean = true;
   activeKeywords = 0;
 
-  subscriptions: Subscription = new Subscription();
+  subscriptionsTerminator$ = new Subject<void>();
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/performance-view/measurements/measurements-picker.component.ts:21]: I would prefer to have a shorter name
just `terminator$` or `destroy$`

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/time-selection/performance-view-time-selection.component.ts [hunk 3]
```diff
@@ -157,8 +158,6 @@ export class PerformanceViewTimeSelectionComponent implements OnInit, OnDestroy
   }
 
   ngOnDestroy(): void {
-    if (this.subscriptions) {
-      this.subscriptions.unsubscribe();
-    }
+    this.subscriptionsTerminator$.next();
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/performance-view/time-selection/performance-view-time-selection.component.ts:161]: Completion missing

**Review by dvladir — CHANGES_REQUESTED** [review]
Check comments

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/measurements/measurements-picker.component.ts (hunk 1 above)
Comment (lucianbaciu) [reply projects/step-frontend/src/lib/modules/timeseries/performance-view/measurements/measurements-picker.component.ts]: @dvladir I did the changes, but it seems to work without takeUntil. are you sure it is needed? maybe it's safer to complete it, but takeUntil will react when a .next() is called on the observable. Do I miss something?

Comment (dvladir) [reply projects/step-frontend/src/lib/modules/timeseries/performance-view/measurements/measurements-picker.component.ts]: @lucianbaciu 
I played around a bit. Yes you're right `takeUntil` completes the source stream, when subject's `next` is invoked.
So `complete` inactivation not required for termination - I was wrong about it

But anyway completion of any subjects, that were declared in the component - is a good practice. I highly recommend to keep it.

**Review by dvladir — APPROVED** [review]

**Review by neogucky — APPROVED** [review]

**Pull request merged**